### PR TITLE
allow extended SWA config

### LINKED ISSUE #1: Allow custom SWA config
We should allow the user to extend the [default SWA config](https://github.com/geoffrich/svelte-adapter-azure-swa/blob/5f4492b9f73b2871c9a62c5f19f11a45b8bffece/files/staticwebapp.config.json) to add custom redirects, auth, etc.

Potentially, allow the user to create a config in the root directly and merge it with the default config. We should error on overriding `navigationFallback` since this is critical for the SSR functionality and endpoints. Other redirects can be added so long as the catch-all redirect for post/put/delete is at the end. Non-routing settings can be copied over as-is.

This file should be named differently so that the deployment picks up the generated config.

### PR DESCRIPTION
**Pull request by biltongza**
Fixes #1.

I hate the name of `ExtendStaticWebAppConfig` but I can't think of anything better at the moment.

Also not 100% certain on the validation logic. I figure it's better to try stop people getting into situations that will leave their app completely broken, but I'm not sure how far down that rabbit hole to go, so I settled for checking if they try override the catch all route or if they specify `navigationFallback` (even though the type says you `navigationFallback` is not valid, since I used a destructure on the extended config, it could still technically happen, maybe there is a better way to do this).

I could see an issue with someone trying to specify auth on the catch all route that this would then reject?

The docs specify an `auth` property (https://docs.microsoft.com/en-us/azure/static-web-apps/configuration#authentication) as well as `forwardingGateway` and `networking` on the config, but these aren't present in the SWA CLI types: https://github.com/Azure/static-web-apps-cli/blob/f2ec90afb5a4d3d4caeb1ffe4f56776a937cfac1/src/swa.d.ts#L130 so I decided not to include them.

I figure I should probably update the readme with a quick example too.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,15 +9,34 @@ import esbuild from 'esbuild';
...
 const ssrFunctionRoute = '/api/__render';
 
+/**
+ * Validate the static web app configuration does not override the minimum config for the adapter to work correctly.
+ * @param config {import('./types/swa').ExtendStaticWebAppConfig} 
+ * */
+function validateExtendConfig(config) {
+	if (config) {
+		if ('navigationFallback' in config) {
+			throw new Error("Cannot override navigationFallback since is used for SSR and endpoints.");
```
Comment (geoffrich) [inline index.js:19]: Single quotes for consistency.

File: index.js [hunk 2]
```diff
@@ -9,15 +9,34 @@ import esbuild from 'esbuild';
...
+		}
+		if (config.routes && config.routes.find(route => route.route === '*')) {
+			throw new Error("Cannot override catch-all route.");
```
Comment (geoffrich) [inline index.js:22]: Single quotes for consistency.

File: index.js [hunk 3]
```diff
@@ -9,15 +9,34 @@ import esbuild from 'esbuild';
...
 		name: 'adapter-azure-swa',
 
 		async adapt(builder) {
+			validateExtendConfig(extendStaticWebAppConfig);
+
 			/** @type {import('./types/swa').StaticWebAppConfig} */
 			const swaConfig = {
+				...extendStaticWebAppConfig,
 				routes: [
+					...(extendStaticWebAppConfig || { routes: [] }).routes,
```
Comment (geoffrich) [inline index.js:39]: I think this will throw if `extendStaticWebAppConfig` is defined but has no routes.

File: index.js [hunk 4]
```diff
@@ -9,15 +9,34 @@ import esbuild from 'esbuild';
...
+		}
+		if (config.routes && config.routes.find(route => route.route === '*')) {
+			throw new Error("Cannot override catch-all route.");
+		}
+	}
+}
+
 /** @type {import('.')} */
-export default function ({ debug = false } = {}) {
+export default function ({ debug = false, extendStaticWebAppConfig = undefined } = {}) {
```
Comment (geoffrich) [inline index.js:28]: Can we default this to `{}` instead of undefined? Otherwise I think line 37 will throw if no extended config is provided.

**Review by geoffrich — CHANGES_REQUESTED** [review]
Thanks for the PR! A few small comments down below.

I don't love the name extendStaticWebAppConfig either, though it's not terrible. Definitely a tricky thing to name. What do you think about customStaticWebAppConfig? Or customSWAConfig? Or just config (might be too generic)?

Docs in the README about this setting would also be good to add. Make sure to note which settings cannot be overridden.

I wonder if in the future we'll need to merge route rules, e.g. if the adapter defines a rule for a given route and the user wants to add a setting to that route, like certain headers or auth. But we can cross that bridge when we come to it.

File: index.js (hunk 1 above)
Comment (biltongza) [reply index.js]: Good catch. Wonder why prettier didn't kick in.

File: index.js (hunk 3 above)
Comment (biltongza) [reply index.js]: Done

File: index.js (hunk 4 above)
Comment (biltongza) [reply index.js]: Good catch, done.

**Comment by geoffrich** [comment]
I put a note in the docs that's basically "use at your own risk." I want to eventually play around with SWA auth + this adapter myself, so I might have some ideas then. But I think that's good enough for now until someone opens an issue.

**Comment by geoffrich** [comment]
Thanks for the PR!

**Pull request merged**